src/test/scala/fte/data/GraphCreator.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ Copyright 2016 Jo Pol
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program. If not, see http://www.gnu.org/licenses/gpl.html dibl
+*/
+package fte.data
+
+import dibl.proto.TilesConfig
+import dibl.{ LinkProps, NewPairDiagram }
+import fte.layout.OneFormTorus
+
+object GraphCreator {
+
+  /**
+    * @param urlQuery parameters for: https://d-bl.github.io/GroundForge/tiles?
+    *                 The patch size must span 3 columns and 2 rows of checker tiles.
+    *                 In ascii-art:
+    *                 +----+----+----+
+    *                 |....|....|....|
+    *                 |....|....|....|
+    *                 +----+----+----+
+    *                 |....|....|....|
+    *                 |....|....|....|
+    *                 +----+----+----+
+    * @return
+    */
+  def fromPairs(urlQuery: String): Option[Graph] = {
+    val config = TilesConfig(urlQuery)
+    val cols = config.patchWidth / 3
+    val rows = config.patchHeight / 2
+    val graph = new Graph(rows, cols)
+    val diagram = NewPairDiagram.create(config)
+
+    def inCenterBottomTile(link: LinkProps): Boolean = {
+      // The top and side margins of the pair diagram may have irregularities.
+      val target = diagram.nodes(link.target)
+      val y = target.y / 15 - 2
+      val x = target.x / 15 - 2
+      y >= rows && x >= cols && x < cols * 2
+    }
+
+    diagram.links.filter(inCenterBottomTile)
+      .foreach { link =>
+        val src = diagram.node(link.source)
+        val dest = diagram.node(link.target)
+        graph.addEdge(
+          dest.x.toInt / 15 - 2 , dest.y.toInt / 15 - 2,
+          src.x.toInt / 15 - 2, src.y.toInt / 15 + 2,
```

Comment:
The calculations in the two lines above scale and shift the pair diagram back to the prototype diagram. With calculations the whiting pattern fails. Without the calculations, the whiting pattern succeeds but the braid pattern fails and the pinwheel is rendered out of focus. I don't understand why these corrections make or break your code. See the zips for the logging and rendered diagrams.
[no-calc.zip](https://github.com/d-bl/GroundForge/files/3921225/no-calc.zip)
[with-calc.zip](https://github.com/d-bl/GroundForge/files/3921226/with-calc.zip)